src/metabase/lib/field.cljc

```diff
@@ -787,16 +787,16 @@
 (mu/defn ^:private remapped-field :- [:maybe ::lib.schema.metadata/column]
   [metadata-providerable :- ::lib.schema.metadata/metadata-providerable
    column                :- ::lib.schema.metadata/column]
-  (when-let [remap-field-id (get-in column [:lib/external-remap :field-id])]
-    (lib.metadata/field metadata-providerable remap-field-id)))
+  (or
+    (when-let [name-field-id (:name-field column)]
```

Comment:
This is a hack. There is no `name-field` property on `column` that returns the field id. The actual property has the whole field. This becomes an id only because of FE normalization. I think we should do what we do with `fk-target-field-id` - MBQL lib doesn't rely on the normalized `target` but the actual BE property. Maybe we could add `name-field-id` to the batch hydration method for `:name_field` on the BE and use this new property instead.